### PR TITLE
Adds support for continent specific GeoIP2-City databases

### PR DESCRIPTION
**Pull request by fs1**
Since I use the GeoIP2-City-Europe Database from maxmind I needed to modify the files LocationProvider/GeoIp2/Php.php and LocationProvider/GeoIp2.php in the GeoIp2 Plug-in. After modification the GeoIP file was recognized an worked like expected.
Please consider implementing this patch.

### CODE REVIEW

**Comment by sgiehl** [comment]
Guess the database is available for other continents as well? Maybe we should add them then too.

**Comment by fs1** [comment]
Good idea and [yes, it is](https://www.maxmind.com/en/geoip2-city-database-by-continent). Feel free to extend my proposal.

**Comment by sgiehl** [comment]
@fs1 Guess you don't have purchased or have access to any of the other continent dbs? Would be good to know how their naming is. Otherwise it's hard to add their names

**Comment by sgiehl** [comment]
@fs1  found a page where it's mentioned. Would you mind adjusting your PR, so the other continents are listed as well?

https://dev.maxmind.com/geoip/geoip2/geoip2-city-country-csv-databases/
> For the city subset by continent databases, the {Content} is one of “City-Africa”, “City-Asia-Pacific”, “City-Europe”, “City-North-America”, or “City-South-America”.

**Comment by sgiehl** [comment]
@fs1 thanks for updating. Are you sure there are continent separated dbs for GeoLite2? Did only find some for the paid version...

**Comment by fs1** [comment]
well. guess you're right. 
On https://www.maxmind.com/en/geoip2-city-database-by-continent they write
> Subsets of the GeoIP2 City database are available for individual continents.

This explanation from https://dev.maxmind.com/geoip/geoip2/geoip2-city-country-csv-databases/ ist some kind of contradictory
> The zip file itself is named {Product}-{Content}-CSV_latest.zip. The {Product} is either “GeoIP2” or “GeoLite2”. The {Content} refers to the particular product name, such as “City” or “Country”. For the city subset by continent databases, the {Content} is one of “City-Africa”, “City-Asia-Pacific”, “City-Europe”, “City-North-America”, or “City-South-America”. 

Im going to update the PR again.

File: plugins/GeoIp2/LocationProvider/GeoIp2/Php.php [hunk 1]
```diff
@@ -273,7 +283,7 @@ public function getInfo()
             . '</a>';
 
         $availableDatabaseTypes = array();
-        if (self::getPathToGeoIpDatabase(['GeoIP2-Enterprise.mmdb', 'GeoIP2-City.mmdb', 'GeoLite2-City.mmdb']) !== false) {
+        if (self::getPathToGeoIpDatabase(['GeoIP2-City.mmdb', 'GeoIP2-City-Africa.mmdb', 'GeoIP2-City-Asia-Pacific.mmdb', 'GeoIP2-City-Europe.mmdb', 'GeoIP2-City-North-America.mmdb', 'GeoIP2-City-South-America.mmdb', 'GeoIP2-Enterprise.mmdb', 'GeoIP2-Country.mmdb', 'GeoLite2-City.mmdb', 'GeoLite2-Country.mmdb']) !== false) {
```
Comment (sgiehl) [inline plugins/GeoIp2/LocationProvider/GeoIp2/Php.php]: The `Country` versions shouldn't be added here, as its the detection if the city level is available.

**Review by sgiehl — APPROVED** [review]

**Comment by sgiehl** [comment]
@fs1  Thanks again for your contribution. Will merge it now, so it will be included in the next release.

**Pull request merged**